keras_nlp/samplers/sampler.py

```diff
@@ -109,7 +101,7 @@ def cond(prompt, state, index):
         def body(prompt, state, index):
             # Compute the softmax distribution for the next token.
             logits, state = next(prompt, state, index)
-            probabilities = keras.activations.softmax(logits)
+            probabilities = keras.activations.softmax(logits / self.temperature)
```

Comment:
I think to make this work you for all samplers, you will need to do a little more here...

 - Add an `__init__` on this base class, take in `temperature=1.0`.
 - Add `self.temperatue` to the config on the base class.
 - In all subclasses, take in a **kwargs as part of init and pass that to super class.